src/pymodaq_utils/serialize/serializer.py

```diff
@@ -254,6 +254,104 @@ def deserialize(bytes_str: bytes) -> Tuple[List[SERIALIZABLE], bytes]:
         return list_obj, remaining_bytes
 
 
+class TupleSerializeDeserialize(SerializableBase):
+    @staticmethod
+    def serialize(tuple_object: Tuple) -> bytes:
+        """ Convert a tuple of objects into a bytes message together with the info to convert it back
+
+        Parameters
+        ----------
+        tuple_object: tuple
+            the tuple could contain whatever objects are registered in the SerializableFactory
+
+        Returns
+        -------
+        bytes: the total bytes message to serialize the tuple of objects
+
+        Notes
+        -----
+
+        The bytes sequence is constructed as:
+        * the length of the tuple
+
+        Then for each object:
+        * use the serialization method adapted to each object in the tuple
+        """
+        if not isinstance(tuple_object, tuple):
+            raise TypeError(f'{tuple_object} should be a tuple, not a {type(tuple_object)}')
+        return ListSerializeDeserialize().serialize(list(tuple_object))
+
+    @staticmethod
+    def deserialize(bytes_str: bytes) -> Tuple[Tuple[SERIALIZABLE], bytes]:
+        """Convert bytes into a tuple of objects
+
+        Convert the first bytes into a tuple reading first information about the tuple elt types, length ...
+
+        Returns
+        -------
+        tuple: the decoded tuple
+        bytes: the remaining bytes string if any
+        """
+        list_object, remaining_bytes = ListSerializeDeserialize().deserialize(bytes_str)
+        return tuple(list_object), remaining_bytes
+
+
+class DictSerializeDeserialize(SerializableBase):
+    @staticmethod
+    def serialize(dict_object: dict) -> bytes:
```

Comment:
```suggestion
    def serialize(dict_object: dict[SERIALIZABLE, SERIALIZABLE]) -> bytes:
```
Alternatively, it could be `dict[str, SERIALIZABLE]`.

here, the first one is the key type, the second one the value type.